test/e2e/plain_provisioner_test.go

```diff
@@ -544,6 +544,135 @@ var _ = Describe("plain provisioner bundleinstance", func() {
 		})
 	})
 
+	When("a BundleInstance is dependent on another BundleInstance", func() {
```

Comment:
nit: does it make it more clear to have the test name be slightly more specific ie `a BundleInstance is dependent on APIs provided by another BundleInstance?` Since BIs are independent of one another. I didn't fully understand the context of the test until I looked at the manifests in the testdata directory. 

Or is the intention here that this test is generalizable to all cases? Like for example if one BI needs a namespace to install a resource, and another BI comes along and installs that namespace, and then the first BI succeeds?

Reply:
What I had in mind was closer to the latter, and we can easily emulate trying to instantiate a Bundle that contains a namespaced-scoped resource and explicitly defines a metadata.Namespace that doesn't exist on-cluster, and then create the providing Bundle/BI resource that rolls out that namespace.